addons/udes_priorities/models/priority.py

```diff
@@ -84,45 +105,86 @@ def _has_duplicate_names_for_same_pick_type(self):
     @api.one
     def _check_for_duplicate_names_for_same_pick_type(self):
         if self._has_duplicate_names_for_same_pick_type() > 0:
-            raise UserError("Name must be unique for the picking types it is shown on")
+            raise UserError("Name must be unique for the transfer types it is shown on")
 
-    @api.multi
-    def _number_of_oustanding_pickings(self):
-        Picking = self.env["stock.picking"]
-        self.ensure_one()
-        return Picking.search_count(
-            [("priority", "=", self.reference), ("state", "not in", ("cancel", "done")),]
+    def _is_trusted_user(self):
+        return self.env.uid == SUPERUSER_ID or self.env.user.has_group(
+            "udes_security.group_trusted_user"
         )
 
-    @api.multi
-    def _check_for_oustanding_pickings(self):
+    def _get_outstanding_pickings(self, reference_id, pick_type_id=None):
+        Picking = self.env["stock.picking"]
```

Comment:
At the import `from odoo.osv import expression`
`search_domain = [("priority", "=", reference_id),("state", "not in", ("cancel", "done"))]`
`if pick_type_id:
    search_domain = expression.AND([search_domain, [("picking_type_id", "=", pick_type_id)]])`
`return Picking.search_count(search_domain)`